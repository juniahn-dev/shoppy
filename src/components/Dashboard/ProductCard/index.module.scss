.container {
  display: flex;
  flex-direction: column;

  overflow: hidden;
  border-radius: 1rem;

  box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
}

.image {
  width: 100%;
  height: 100%;
}

.productInfo {
  padding: 0.5rem 0.7rem;
}

.productTitle {
  display: flex;
  justify-content: space-between;
  align-items: center;

  margin-bottom: 0.5rem;

  font-weight: bold;
}

.title {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}
