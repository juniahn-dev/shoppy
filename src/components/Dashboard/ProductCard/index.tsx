import { IProductListProps } from '@/types/firebaseTypes';
import ImageComponent from '@/components/Common/Image';
import styles from './index.module.scss';

interface IProductCardProps {
  product: IProductListProps;
}

const ProductCard: React.FC<IProductCardProps> = ({ product }) => {
  return (
    <div className={styles.container}>
      <ImageComponent className={styles.image} src={product.image} />
      <div className={styles.productInfo}>
        <div className={styles.productTitle}>
          <div className={styles.title}>{product.title}</div>${product.price}
        </div>
        {product.category}
      </div>
    </div>
  );
};

export default ProductCard;
