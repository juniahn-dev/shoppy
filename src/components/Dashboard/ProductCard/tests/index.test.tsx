import { render, screen } from '@testing-library/react';

import ProductCard from '..';
import { fakeProduct } from '@/test/datas';
import renderer from 'react-test-renderer';

describe('ProductCard', () => {
  const { image, title, price, category } = fakeProduct;

  // use snapshot
  it('renders flex type card', () => {
    const component = renderer.create(<ProductCard product={fakeProduct} />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  // not use snapshot
  it('renders product cart', () => {
    render(<ProductCard product={fakeProduct} />);

    const productImg = screen.getByRole('img') as HTMLImageElement;

    expect(productImg.src).toBe(`${image}/`);
    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(`$${price}`)).toBeInTheDocument();
    expect(screen.getByText(category)).toBeInTheDocument();
  });
});
