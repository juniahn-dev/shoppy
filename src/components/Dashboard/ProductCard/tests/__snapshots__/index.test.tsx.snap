// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders flex type card 1`] = `
<div
  className="container"
>
  <img
    alt="image"
    className="image"
    height="1000"
    src="https://image"
    width="1000"
  />
  <div
    className="productInfo"
  >
    <div
      className="productTitle"
    >
      <div
        className="title"
      >
        gorgeous shirt
      </div>
      $
      124000
    </div>
    female
  </div>
</div>
`;
